### ISSUE TITLE
crash, when an existing logged in user log in again

### ISSUE THREAD

**Issue by Festus1965** [MultiCraft/MultiCraft2#16]
##### Minetest version
```
multicraft 2.0.0 (Linux)
```

##### OS / Hardware
Operating system: Ubuntu 18.04
CPU: I7-4770

##### Summary
It seams, someone try to login with same name then another existing user, or try to login with same own nick same time.
As I was not there at this time, I couldn't see my own report on admin client get a pre-join info who try to login. 
Happened now 2 or 3 times, and as I know under 5.2.0 that made NOT a crash, as sometime I did the same mistake, try login with Thailand, but was logged in already.
(translation short:
It seems as if someone is trying to copy another user who is already logged in or to log in twice with his own user name)

##### Steps to reproduce
2021-09-11 18:36:27: ERROR[Server]: TOSERVER_CLIENT_READY stage 2 client init failed peer_id=150
...
2021-09-11 23:55:31: ERROR[Main]: Failed to finalize m_stmt_create: UNIQUE constraint failed: auth.name
2021-09-11 23:55:32: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod '' in callback createAuth(): : UNIQUE constraint failed: auth.name
2021-09-11 23:55:32: ERROR[Main]: stack traceback:
2021-09-11 23:55:32: ERROR[Main]: 	[C]: at 0x5567448a6d00

**Comment by MoNTE48** [comment]
Tip: use the LevelDB backend for the auth database. SQLite3 has a lot of problems in Minetest.

**Comment by luk3yx** [comment]
It looks like this has happened on Minetest as well (https://github.com/minetest/minetest/issues/9498). That issue was opened before 5.2.0 was released so it's strange that you haven't(?) encountered it on 5.2.0.

**Comment by Festus1965** [comment]
never before !
I and I often forgot that Siriphorn or Thailand is logged on server, and I tried on my PC,
or the kids take wrong PC of 5 and forget they are in on another ...
never crash as of this before

but remember: my server kept under 5.2.0 as from 5.3.0 started with incompatible changes, that crashed a start ...

**Comment by Festus1965** [comment]
> Tip: use the LevelDB backend for the auth database. SQLite3 has a lot of problems in Minetest.

I have already PostgreSQL for player and world, still not possible to use it for auth also,
a third db on the server is not my favorite solution, but I test, as just happened again

**Comment by Festus1965** [comment]
so I was looking around:
* in my .conf I had sqlite_synchronius = 0 (set now to 1 for nect resatart / crash)
* in Multicraft2/src/defaultsettings.ssp I found settings->setDefault("sqlite_synchronous", "2"); 

so which one is than the one active ? 0 or 2

I prepared change to both settings to 1, as long there is no other responce

**Comment by Festus1965** [comment]
so after 5 days:
at the auth it is a problem ALSO from Sqlite3, after I migrated to levelDB - not anymore,
BUT
it happened also in until now 3? more mods, that during the 'work' needed data from player was nil.

That could be as of changes from 5.2.0 to 5.4.1 OR from MC itself, but so far I solved it with most
check the value before the crash line for nil, and avoid then to go ahead.

Sure, my server has never seen 1000 new gamer at one day ... and no oher MT server has for sure, as not reachable for the most clients ...maybe, mt is not build for mass ! adding accounts
